File: solutions/TypeScript/[0389_Easy] Find the Difference/findTheDifference.ts
function findTheDifference(s: string, t: string): string {
  let sum = 0;

  for (let i = 0; i < Math.max(s.length, t.length); i++) {
    if (i < s.length) {
      sum -= s.charCodeAt(i);
    }

    if (i < t.length) {
      sum += t.charCodeAt(i);
    }
  }

  return String.fromCharCode(sum);
}

File: solutions/Java/[0389_Easy] Find the Difference/findTheDifference.java
class Solution {
  public char findTheDifference(String s, String t) {
    int sSum = 0;
    int xSum = 0;

    for (int i = 0; i < Math.max(s.length(), t.length()); i++) {
      if (i < s.length()) {
        sSum ^= s.charAt(i);
      }

      if (i < t.length()) {
        xSum ^= t.charAt(i);
      }
    }

    return (char) (sSum ^ xSum);
  }
}
